x-pack/legacy/plugins/siem/public/containers/detection_engine/signals/api.ts

```diff
@@ -10,8 +10,20 @@ import { throwIfNotOk } from '../../../hooks/api/api';
 import {
   DETECTION_ENGINE_QUERY_SIGNALS_URL,
   DETECTION_ENGINE_SIGNALS_STATUS_URL,
+  DETECTION_ENGINE_INDEX_URL,
+  DETECTION_ENGINE_PRIVILEGES_URL,
 } from '../../../../common/constants';
-import { QuerySignals, SignalSearchResponse, UpdateSignalStatusProps } from './types';
+import {
+  QuerySignals,
+  SignalSearchResponse,
+  UpdateSignalStatusProps,
+  SignalsIndex,
+  SignalIndexError,
+  Privilege,
+  PostSignalError,
+  BasicSignals,
+} from './types';
+import { parseJsonFromBody } from '../../../components/ml/api/throw_if_not_ok';
```

Comment:
nit: May want to move `parseJsonFromBody` to a generic utility as it's no longer specific to `ml/api`